src/main/java/Search/Searcher.java

```diff
@@ -0,0 +1,142 @@
+package Search;
+
+import Print.Printable;
+import Print.IntegerPair;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.TreeMap;
+
+import com.google.re2j.Matcher;
+import com.google.re2j.Pattern;
+
+public class Searcher {
+    private static final String DEFAULT_CHARSET = null; // Indicates platform default for FileUtils
+    private static final int PERIOD_MATCHES_NEWLINES_AND_SPACES = Pattern.DOTALL;
+    private static final String MATCH_INDICATOR = "||";
+    private static final char DELIMITER = ' ';
+
+    public Printable getPrintableSearchResult(Searchable searchable, String search) {
+        String transcript = getTranscript(searchable);
+        String timestamps = getTimestamps(searchable.timestampFile);
+        List<IntegerPair> matches = getMatches(transcript, search);
+        TreeMap<Integer, IntegerPair> transcriptTimestampIndices = mapTranscriptToTimestamps(transcript, timestamps);
+
+        return new Printable(matches, transcript, timestamps, searchable.filename, transcriptTimestampIndices);
+    }
+
+    private TreeMap<Integer, IntegerPair> mapTranscriptToTimestamps(String transcript, String timestamps) {
+        TreeMap<Integer, IntegerPair> transcriptTimestampIndices = new TreeMap<>();
+
+        int timestampStart = 0;
+        int timestampEnd = getNextDelimiterIndex(timestampStart, timestamps);
+
+        // base case for the first transcript word, which does not have a prior space
+        IntegerPair firstTimestamp = new IntegerPair(timestampStart, timestampEnd);
+        transcriptTimestampIndices.put(0, firstTimestamp);
+
+        for (int transcriptIndex = 0; transcriptIndex < transcript.length(); ++transcriptIndex) {
```

Comment:
prefer a functional over imperative style to keep things easy to read.

If you also make changes as per other comments, you could replace this big for loop with something like:
```
private TreeMap<Integer, String> mapTranscriptToTimestamps(String transcriptText, String timestampsText) {
      // kinda ugly hack to make sure 0 is in the list. many other ways you could do this
      List<Integer> transcriptWordIndices = getDelimiterIndices(DELIMITER + transcriptText);
      List<String> timestamps = timestampsText.split(DELIMITER);

      TreeMap<Integer, String> transcriptTimestampIndices = new TreeMap<>();
      IntStream.range(0, timestamps.size()-1).boxed()
              .forEach(i -> transcriptTimestampIndices.put(
                  transcriptWordIndices.get(i),
                  timestamps.get(i)
              ));
      return transcriptTimestampIndices;
  }
```